lib/spack/spack/build_systems/cmake.py

```diff
@@ -284,7 +288,10 @@ class CMakeBuilder(BaseBuilder):
     @property
     def archive_files(self):
         """Files to archive for packages based on CMake"""
-        return [os.path.join(self.build_directory, "CMakeCache.txt")]
+        return [
+            os.path.join(self.build_directory, "CMakeCache.txt"),
+            os.path.join(self.build_directory, "compile_commands.json"),
+        ]
```

Comment:
```suggestion
        files = [
            os.path.join(self.build_directory, "CMakeCache.txt"),
        ]
        
        if cmake.satisfies("@3.5:"):
            os.path.join(self.build_directory, "compile_commands.json"),
```
And this could probably have `if cmake.satisfies(compilation_database_constraints)` to avoid duplicating code with line 99

Reply:
Good idea. I couldn't find a way to describe the requirements as a single spec string, so I added a `_supports_compilation_databases` function in https://github.com/spack/spack/pull/42353/commits/bfda1626b8a8c94586c3376f6392b3979847db8e and used it in both places.